TopPlaces/TopPlaces/Model/Flickr/FlickrPlacesPhotosProvider.m

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) 2016 Lightricks. All rights reserved.
+// Created by Gennadi Iosad.
+
+#import "FlickrPlacesPhotosProvider.h"
+
+#import "FlickrFetcher.h"
+#import "FlickrPhotoInfo.h"
+#import "FlickrPlaceInfo.h"
+#import "FlickrCancellableTask.h"
+
+NS_ASSUME_NONNULL_BEGIN
+@implementation FlickrPlacesPhotosProvider
+
+- (id<CancellableTask>)downloadPlacesWithCompletion:
+        (DownloadPlacesCompleteBlock)downloadCompleteBlock {
+  //download and convert places to array
+  NSURL *url = [FlickrFetcher URLforTopPlaces];
+
+  id<CancellableTask> task =
+  [self downloadDataFromUrl:url
+          withCompletion:^(NSData *data, NSError *error) {
+            NSMutableArray *placesInfo = nil;
+            if (data && !error) {
+              NSDictionary *propertyListResults = [NSJSONSerialization JSONObjectWithData:data
+                                                                                  options:0
+                                                                                    error:NULL];
+              
+              NSArray *placesData = [propertyListResults valueForKeyPath:FLICKR_RESULTS_PLACES];
+              if (placesData) {
+                placesInfo = [[NSMutableArray alloc] initWithCapacity:[placesData count]];
+                for (NSDictionary *d in placesData) {
+                  [placesInfo addObject:[[FlickrPlaceInfo alloc] initWithDictionary:d]];
+                }
+              }
+            }
+            downloadCompleteBlock(placesInfo, error);
```

Comment:
When it comes to completion block you must check that it is not <code>nil</code> before invoking the block, otherwise you'll end up with exception.